### DISCUSSION TITLE
Understanding include_background

### DISCUSSION THREAD

**Discussion by 25benjaminli** [Project-MONAI/tutorials#1632] [General] [answered]
Let's say I have two tensors (1,3,224,224) and (1,3,224,224), representing prediction and target, with sigmoid being applied to the prediction tensor. 3 is the number of classes in the semantic segmentation problem, and (224x224) is the spatial dimensions. I see that there are several parameters to the dice loss and other relevant functions, one of which being include_backgrounds. How would you explain include_backgrounds here? Is it applicable?

I am asking this question because in the monai tutorials, like brats3dsegmentation.ipynb, they use include_backgrounds despite having only 3 channels and no channel to represent the background images.

**Comment by KumoLiu** [comment]
Hi @25benjaminli, thanks for your interest here.
In MONAI, we typically assume that the first channel of the data is the background, so include_background is also an indication of whether or not to consider the first channel.
https://github.com/Project-MONAI/MONAI/blob/33afaefff925dce45b8331e8290f234a4d95fd51/monai/metrics/meandice.py#L40-L41
https://github.com/Project-MONAI/MONAI/blob/33afaefff925dce45b8331e8290f234a4d95fd51/monai/metrics/meandice.py#L269
Hope it helps, thanks.

**Comment by 25benjaminli** [comment]
@KumoLiu thanks for your response. If this is the case, does the tutorial using swin unetr and segresnet use include_background incorrectly? As I mentioned, the outputs of their models are (b, 3, [spatial_dims]) which is being compared to (b, 3, [spatial dims]) label. Or do they just pretend that the "background label" is representing an actual label?
A link is attached below about this. Thanks again. 
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/swin_unetr_brats21_segmentation_3d.ipynb

**Reply by KumoLiu** [reply, marked as answer]
Yes, there are relevant tickets that talk about the name of the `include_background`, would be better to revisit it to avoid misunderstanding here.
https://github.com/Project-MONAI/MONAI/discussions/6915#discussioncomment-7097583
https://github.com/Project-MONAI/MONAI/issues/7056

**Reply by 25benjaminli** [reply]
@KumoLiu Thank you for the response. This makes sense now.
